[arjun2598] iP

arjun2598 opened the pull request:
### MichaelPro

> “The key is not to prioritize what’s on your schedule, but to schedule your priorities.” – Stephen Covey [(source)](https://dansilvestre.com/productivity-quotes/#4_“The_key_is_not_to_prioritize_whats_on_your_schedule_but_to_schedule_your_priorities”_–_Stephen_Covey)

MichaelPro helps you track the things you need to do. It's,
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

All you need to do is,
1. download it from [here](https://github.com/arjun2598/ip/releases).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 🤖 

And it is **FREE!**

Features: 
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practise Java too. Here's the `main` method:
```java

public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

arjun2598 committed:
Rename chatbot, add greeting and exit

arjun2598 committed:
Users can give chatbot commands

arjun2598 committed:
Add user inputs to a list, list them back when user requests

arjun2598 committed:
Add Task class

arjun2598 committed:
Mark tasks as done or change back to not done

arjun2598 committed:
Add function to handle user commands

arjun2598 committed:
Add ToDo task

arjun2598 committed:
Add Deadline task

arjun2598 committed:
Add Event task

arjun2598 committed:
Fix output for listing tasks

arjun2598 committed:
Add automated text UI testing

arjun2598 committed:
Add MichaelException class

arjun2598 committed:
Handle empty inputs after command

arjun2598 committed:
Add ability to delete task

arjun2598 committed:
Add save file to save tasks

arjun2598 committed:
Read tasks in save file and load into chatbot

arjun2598 committed:
Write to save file correctly

arjun2598 committed:
Handle cases where either save file does not exist or directories do not exist

arjun2598 committed:
Edit code and comments

arjun2598 committed:
Merge branch 'branch-Level-7'

arjun2598 committed:
Store dates for deadline tasks as LocalDate

arjun2598 committed:
Handle invalid format of date

arjun2598 committed:
Merge branch 'branch-Level-8'

arjun2598 committed:
Add classes to handle each part of the program

arjun2598 committed:
Merge branch 'branch-A-MoreOOP'

arjun2598 committed:
Put all classes into package michael

arjun2598 committed:
Merge branch 'branch-A-Packages'

superb-sushi commented on src/main/java/michael/Ui.java:
```diff
@@ -0,0 +1,24 @@
+        System.out.println(error);
+    }
+
+    public void feedback(String feedback) {
```
Would it be better to make this method name a verb?
```suggestion
    public void giveFeedback(String feedback) {
```

superb-sushi commented on src/main/java/michael/Storage.java:
```diff
@@ -0,0 +1,109 @@
+    }
+
+    // Method converts task into text format required for save file
+    public String convert(Task task) {
```
Could the name of this method be changed to give the reader a better understanding of what it does?
```suggestion
    public String convertTaskToString(Task task) {
```

superb-sushi commented on src/main/java/michael/Task.java:
```diff
@@ -0,0 +1,35 @@
+
+public class Task {
+    private final String taskName;
+    private boolean done;
```
Would it be better to name the variable with another name that would immediately signal it as a boolean variable?
```suggestion
    private boolean isDone;
```

superb-sushi commented on src/main/java/michael/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+
+    @Override
+    public String toString() {
+        String formatted_deadline = this.deadline.format(DateTimeFormatter.ofPattern("d MMM yyyy"));
```
Could the variable name be changed to follow camelCase standards?
```suggestion
        String formattedDeadline = this.deadline.format(DateTimeFormatter.ofPattern("d MMM yyyy"));
```

superb-sushi reviewed (commented):
Overall a job well done! A few minor coding quality violations here and there, but otherwise, code appears to be good and functioning!

arjun2598 committed:
Merge branch 'add-gradle-support'

arjun2598 committed:
Build and run using gradle

arjun2598 committed:
Test Storage, TaskList and ToDo classes

arjun2598 committed:
Create JAR file

arjun2598 committed:
Add JavaDoc comments to non-private methods

arjun2598 committed:
Edit code to match coding standard

arjun2598 committed:
Add find command for chatbot

arjun2598 committed:
Merge branch 'branch-A-JavaDoc'

arjun2598 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/michael/Storage.java
#	src/main/java/michael/Ui.java

arjun2598 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/michael/Parser.java

mingyang143 commented on src/main/java/michael/Michael.java:
```diff
@@ -0,0 +1,67 @@
+    private Storage storage; // Handles loading and saving of tasks
+    private TaskList tasks; // Handles adding, retrieval and deletion of tasks
+    private Parser parser; // Processes user commands
+    private final String PATH = "./data/save.txt"; // File path for save file of tasks
```
Constant is labelled correctly.

mingyang143 commented on src/main/java/michael/Parser.java:
```diff
@@ -0,0 +1,133 @@
+                throw new MichaelException("Deadline should be in format YYYY-MM-DD");
+            }
+
+            Deadline curr = new Deadline(parts[0], deadline);
```
Should commands be abstracted away into a separate command class so as to leave the Parser class with purely handling inputs from users?

mingyang143 commented on src/main/java/michael/Parser.java:
```diff
@@ -0,0 +1,133 @@
+                list = list.concat(elem);
+            }
+            return list.substring(0, list.length() - 1); // substring to remove last line break
+        } else { // invalid command
```
Consistent detailed comments done to help with understanding code. Well done!

mingyang143 commented on src/main/java/michael/Michael.java:
```diff
@@ -0,0 +1,67 @@
+        new Michael().run();
+    }
+
+    /**
```
JavaDocs done in right format.

mingyang143 commented on src/main/java/michael/Task.java:
```diff
@@ -0,0 +1,48 @@
+
+public class Task {
+    private final String taskName;
+    private boolean isDone;
```
Boolean named correctly.

mingyang143 commented on src/test/java/michael/StorageTest.java:
```diff
@@ -0,0 +1,21 @@
+import org.junit.jupiter.api.Test;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class StorageTest {
```
Should there be more tests for all the methods inside Storage and testing all the scenarios for each method.

mingyang143 commented on src/test/java/michael/StorageTest.java:
```diff
@@ -0,0 +1,21 @@
+    }
+
+    @Test
+    public void Test2() {
```
The test name should comply to "featureUnderTest_testScenario_expectedBehavior()" as seen in the java coding standards.

mingyang143 commented on src/main/java/michael/Parser.java:
```diff
@@ -0,0 +1,133 @@
+
+            String deadline = parts[1].substring(3);
+            if (deadline.split("-").length < 3) { // invalid format of date
+                throw new MichaelException("Deadline should be in format YYYY-MM-DD");
```
Excellent! Error message is detailed in helping users use the application smoothly.

mingyang143 commented on src/main/java/michael/Storage.java:
```diff
@@ -0,0 +1,129 @@
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Scanner;
```
Imports done well as it is specific and not *.

mingyang143 commented on src/main/java/michael/Task.java:
```diff
@@ -0,0 +1,48 @@
+package michael;
+
+public class Task {
```
Should Task class be made abstract?

mingyang143 reviewed (commented):
Generally well down. There are just some nits like changing the name of the methods in the test files to the right naming format. Other than that, the rest is excellent. Good Job!

arjun2598 committed:
Add checkstyle capability

arjun2598 committed:
Merge branch 'branch-A-CheckStyle'

arjun2598 committed:
Add basic gui

arjun2598 committed:
Add new images for user and Michael

arjun2598 committed:
Edit comment

arjun2598 committed:
Create jar file using gradle

arjun2598 committed:
Add assertions

Tasks to mark, unmark and delete should be within the boundaries of the
list. Format of deadline and event tasks should also be appropriate.

Check for the above using assertions to ensure the input matches what is
expected by the corresponding command.

arjun2598 committed:
Add command classes

Parse method was too long. Splitting the logic into different command
classes helps to abstract out background work needed.

arjun2598 committed:
Check input within command classes

arjun2598 committed:
Add bye command

arjun2598 committed:
Fix bugs with GUI

arjun2598 committed:
Clean up code

arjun2598 committed:
Merge pull request #3 from arjun2598/branch-A-CodeQuality

Improve code quality

arjun2598 committed:
Merge branch 'master' of https://github.com/arjun2598/ip

arjun2598 committed:
Merge branch 'master' into branch-A-Assertions

# Conflicts:
#	src/main/java/michael/Parser.java

arjun2598 committed:
Merge pull request #2 from arjun2598/branch-A-Assertions

Add assertions

arjun2598 committed:
Sort tasks in alphabetical order

arjun2598 committed:
Merge branch 'branch-BCD-Extension'

arjun2598 committed:
Add more error handling

arjun2598 committed:
Credit sources of reuse

arjun2598 committed:
Add user guide information

arjun2598 committed:
Fix Gui bugs with starting and exiting

arjun2598 committed:
Add class javadoc comments

arjun2598 committed:
Add javadoc comments for methods

arjun2598 committed:
Fix naming of tests

arjun2598 committed:
Add minor edits

arjun2598 committed:
Fix bug with generated jar file

arjun2598 committed:
Fix bugs with jar file

arjun2598 committed:
Update User Guide
